azurelinuxagent/common/osutil/openwrt.py

```diff
@@ -0,0 +1,157 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+import os
+import re
+import azurelinuxagent.common.logger as logger
+import azurelinuxagent.common.utils.shellutil as shellutil
+import azurelinuxagent.common.utils.fileutil as fileutil
+from azurelinuxagent.common.osutil.default import DefaultOSUtil
+from azurelinuxagent.common.utils.networkutil import NetworkInterfaceCard
+
+class OpenWRTOSUtil(DefaultOSUtil):
+
+    def __init__(self):
+        super(OpenWRTOSUtil, self).__init__()
+        self.agent_conf_file_path = '/etc/waagent.conf'
+        self.dhclient_name = 'udhcpc'
+        self.ip_command_output = re.compile('^\d+:\s+(\w+):\s+(.*)$')
+        self.jit_enabled = True
+        
+    def eject_dvd(self, chk_err=True):
+        logger.warn('eject is not supported on OpenWRT')
+
+    def useradd(self, username, expiration=None, comment=None):
+        """
+        Create user account with 'username'
+        """
+        userentry = self.get_userentry(username)
+        if userentry is not None:
+            logger.info("User {0} already exists, skip useradd", username)
+            return
+
+        if expiration is not None:
+            cmd = "useradd -m {0} -s /bin/ash -e {1}".format(username, expiration)
+        else:
+            cmd = "useradd -m {0} -s /bin/ash".format(username)
+        
+        if not os.path.exists("/home"):
+            os.mkdir("/home")
+
+        if comment is not None:
+            cmd += " -c {0}".format(comment)
+        retcode, out = shellutil.run_get_output(cmd)
+        if retcode != 0:
+            raise OSUtilError(("Failed to create user account:{0}, "
+                               "retcode:{1}, "
+                               "output:{2}").format(username, retcode, out))
+
+    def get_dhcp_pid(self):
+        cmd = "pidof {0}".format(self.dhclient_name)
+        ret= shellutil.run_get_output(cmd)
+        return ret[1] if ret[0] == 0 else None
+
+    def get_nic_state(self):
+        """
+        Capture NIC state (IPv4 and IPv6 addresses plus link state).
+
+        :return: Dictionary of NIC state objects, with the NIC name as key
+        :rtype: dict(str,NetworkInformationCard)
+        """
+        state = {}
+        status, output = shellutil.run_get_output("ip -o link", chk_err=False, log_cmd=False)
+        """
+        1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN qlen 1000\ link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
```

Comment:
Properly escape the backslash to avoid future flake8 errors.

Reply:
I will remove the example, that will make it safer

Reply:
Fix have been done